src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java

```diff
@@ -117,8 +117,8 @@ public static DatadogGlobalConfiguration getDatadogGlobalDescriptor() {
     public static DatadogJobProperty getDatadogJobProperties(@Nonnull Run r) {
         try {
             return (DatadogJobProperty) r.getParent().getProperty(DatadogJobProperty.class);
-        } catch (NullPointerException e) {
-            // It can only throw a NullPointerException when running tests
+        } catch (Exception e) {
```

Comment:
Is it really true that "It can only throw a **NullPointerException** when running tests" and "It can only throw an **Exception** when running tests" are interchangeable?

Same goes for the other changes in this class.
I think blindly suppressing whatever problem was caught with a noop is a dangerous practice.
If something goes wrong with this code, we will have no way of knowing it, as the exception will be silently discarded

Reply:
Agreed; looking more carefully at the return, both r and r.getParent() are already covered by a @Nonnull check. As such, I will just return `(DatadogJobProperty) r.getParent().getProperty(DatadogJobProperty.class);`.